packages/python/accern_xyme/accern_xyme.py

```diff
@@ -2618,11 +2629,33 @@ def _clear_upload(self) -> None:
             raise ValueError("tmp_uri is None")
         self._perform_upload_action("clear", {"uri": uri}, fobj=None)
 
+    def _upload_part(
+            self,
+            buff: bytes,
+            begin: int,
+            offset: int,
+            total_chunks: int,
+            total_size: int,
+            size_processed: Dict[str, int],
+            print_progress: Callable[..., Any]) -> None:
+        print_progress(size_processed["size_processed"] / total_size, False)
+        assert self._tmp_uri is not None
+        new_size = self._append_upload(
+            self._tmp_uri,
+            BytesIO(buff),
+            offset)
+        size_processed["size_processed"] += new_size
+        if (new_size != len(buff) or (new_size != FILE_UPLOAD_CHUNK_SIZE)) \
+                and (offset != total_chunks - 1):
```

Comment:
This doesn't make sense

Reply:
yeah so how can we ensure that the file sizes are all proper but since the last chunk might not have the same size as the other chunks.. still trying to frame a logic that checks if the size of chunks expect the last have all uploaded content of this size (`FILE_UPLOAD_CHUNK_SIZE`)